Maple2.Server.Game/Commands/PlayerCommand.cs

```diff
@@ -221,4 +223,46 @@ private void Handle(InvocationContext ctx, int points, short rank) {
             }
         }
     }
+    private class CurrencyCommand : Command {
+        private readonly GameSession session;
+
+        public CurrencyCommand(GameSession session) : base("currency", "Add currency to player.") {
+            this.session = session;
+
+            var currency = new Argument<string>("currency", "Type of currency to add: meso, meret, valortoken, treva, rue, havifruit, reversecoin, mentortoken, menteetoken, starpoint, mesotoken.");
+            var amount = new Argument<long>("amount", "Amount of currency to add.");
+
+            AddArgument(currency);
+            AddArgument(amount);
+            this.SetHandler<InvocationContext, string, long>(Handle, currency, amount);
+        }
+
+        private void Handle(InvocationContext ctx, string currency, long amount) {
+            try {
+                switch (currency.ToLower()) {
+                    // Handling meso and meret separately because they are not in the CurrencyType enum.
+                    case "meso":
+                        session.Currency.Meso += amount;
+                        break;
+                    case "meret":
+                        session.Currency.GameMeret += amount;
+                        break;
+                    default:
+                        if (TryParseCurrencyType(currency, out CurrencyType parsedCurrencyType)) {
+                            session.Currency[parsedCurrencyType] += amount;
+                        } else {
+                            throw new ArgumentException("Invalid currency type.");
+                        }
+                        break;
```

Comment:
I think we shouldn't return an exception if someone typos. It should just error to the player. We can also just remove the TryParseCurrencyType function as that's really just Enum.TryParse with more steps

```suggestion
                        if (!Enum.TryParse(currency, out CurrencyType currencyType)) {
                            ctx.Console.Error.WriteLine($"Failed to parse currency type: {currency}");
                            ctx.ExitCode = 1;
                            return;
                        }
                        session.Currency[currencyType] += amount;
                        break;
```